apps/teams-test-app/src/components/LocationAPIs.tsx

```diff
@@ -37,9 +51,22 @@ const ShowLocation = (): React.ReactElement =>
           throw new Error('latitude and longitude are required');
         }
       },
-      submit: async locationProps => {
-        await location.showLocation(locationProps);
-        return 'Completed';
+      submit: {
+        withPromise: async locationProps => {
+          await location.showLocation(locationProps);
+          return 'Completed';
+        },
+        withCallback: (locationProps, setResult) => {
+          const callback = (error: SdkError, status: boolean): void => {
+            if (error) {
+              setResult(JSON.stringify(error));
+            } else {
+              setResult('Status: ' + status);
```

Comment:
If we want to reuse the same E2E scenario tests for both, without having to re-write every scenario, we need to make sure the callback and promise paths result in the very same thing written out to the browser. In this case, you're returning `'Completed'` for promise and `'Status :' + status` for callback, which means we'd need two separate tests, because they wouldn't be able to assert on the value. 

I think it would be better to keep these in sync.